internal/envoy/bootstrap.go

```diff
@@ -85,9 +87,65 @@ func Bootstrap(c *BootstrapConfig) *bootstrap.Bootstrap {
 		},
 	}
 
+	if c.GrpcClientCert != "" || c.GrpcClientKey != "" || c.GrpcCABundle != "" {
+		// If one of the two TLS options is not empty, they all must be not empty
+		if !(c.GrpcClientCert != "" && c.GrpcClientKey != "" && c.GrpcCABundle != "") {
+			log.Fatal("You must supply all three TLS parameters - --envoy-cafile, --envoy-cert-file, --envoy-key-file, or none of them.")
+		}
+		b.StaticResources.Clusters[0].TlsContext = grpcUpstreamTLSContext(c.GrpcCABundle, c.GrpcClientCert, c.GrpcClientKey)
+	}
+
 	return b
 }
 
+func grpcUpstreamTLSContext(cafile string, certfile string, keyfile string) *auth.UpstreamTlsContext {
```

Comment:
Suggestion; it might be better to call this somethign like `UpstreamFileTLSContext` or something to make it clear that it returns an UpstreamTLSContext that expects files to be present on disk, not SDS